src/Framework/PaymentGateways/PaymentGateway.php

```diff
@@ -148,30 +185,118 @@ public function cancelSubscription(Subscription $subscription)
     }
 
     /**
+     * @unreleased
      * @inheritDoc
+     * @throws Exception
      */
     public function canSyncSubscriptionWithPaymentGateway(): bool
     {
-        return $this instanceof SubscriptionTransactionsSynchronizable
-            || $this->subscriptionModule->canSyncSubscriptionWithPaymentGateway();
+        if ($this->subscriptionModule) {
+            return $this->subscriptionModule->canSyncSubscriptionWithPaymentGateway();
+        }
+
+        throw new Exception(esc_html__('Method has not been implemented yet.', 'give'));
```

Comment:
Just use `__()` here. We don't want to encourage HTML in exceptions. Same below and elsewhere.

Reply:
@JasonTheAdams The `esc_html__` function prevents HTML output on screen. That means if they add HTML in overwriting text, it should not output as is but rather escape before outputting it. `__()` output whatever overwritten text. For this reason, I am using  `esc_html__`.